### ISSUE TITLE
Unable to authorize organization

### ISSUE THREAD

**Issue by asbjornu** [appveyor/ci#801]
I've gone through the entire OAuth configuration labyrinth in the @HtmlUnit organization as well as on my account here on GitHub and enabled "AppVeyor" and "AppVeyor CI" (why are there two different applications, by the way?) everywhere, but I'm still unable to log in to AppVeyor under my HtmlUnit organization.

The weirdest thing is that [AppVeyor is building NHtmlUnit](https://ci.appveyor.com/project/asashour/nhtmlunit), but under @asashour's user and not @HtmlUnit. What's up?

**Comment by asashour** [comment]
I can delete the AppVeyor NHtmlUnit project, and you can readd under 'HtmlUnit' organization.

Please confirm.

**Comment by asbjornu** [comment]
@asashour Thanks! I've tried to click "Add" for "NHtmlUnit" under the "HtmlUnit" oragnization in AppVeyor, but then [a personal NHtmlUnit project underneath my username](https://ci.appveyor.com/project/asbjornu/nhtmlunit) was created.

You can delete the build under your user, but I'm either way unable to log in to AppVeyor under the HtmlUnit organization.

**Comment by asashour** [comment]
Done, deleted the project under my name.

**Comment by asbjornu** [comment]
@asashour :+1:

**Comment by asbjornu** [comment]
@asashour :+1: By the way, do you have access to log in to the HtmlUnit organization on AppVeyor? It should prompt you with this when logging in, and then you should find "HtmlUnit" in the "- select account -" dropdown menu:

![screen shot 2016-05-12 at 14 17 58](https://cloud.githubusercontent.com/assets/12283/15214287/70e3cf8a-184c-11e6-9ccd-d04daa2d4a55.png)

**Comment by asashour** [comment]
No, I have simple access. Please go ahead with AppVeyor, this area is of no much interest to me.

**Comment by asbjornu** [comment]
@asashour I will, thanks for your help! :smiley:

**Comment by FeodorFitsner** [comment]
This guide maybe helpful too: http://www.appveyor.com/docs/team-setup

**Comment by asbjornu** [comment]
@FeodorFitsner I've read that and it doesn't help. Logging in with my GitHub account does not make "HtmlUnit" available, even though I'm an **owner** in that organization. What can I do to have AppVeyor recognize my (new) role in the HtmlUnit organization and allow me to log in to that team?

**Comment by FeodorFitsner** [comment]
There might be gotchas with multiple GitHub teams assigned: http://www.appveyor.com/docs/team-setup#github-integration-notes

**Comment by asbjornu** [comment]
@FeodorFitsner Nothing in that article makes it clear to me how I'm supposed to be able to sign in to AppVeyor with the NHtmlUnit organization. Can you please elaborate?

**Comment by FeodorFitsner** [comment]
You are signing in with your own GitHub account, not organization - GitHub organization is not an authentication entity. When you first sign up with "GitHub" button AppVeyor account is named after your GitHub username. However, you can decide to make it "organizational" account, rename it to be the same as GitHub org and add collaborators.

**Comment by asbjornu** [comment]
@FeodorFitsner That makes absolutely no sense at all. When I sign in to AppVeyor with my GitHub account, I can choose to log in with these 3 organizations:

![screen shot 2016-05-26 at 20 39 15](https://cloud.githubusercontent.com/assets/12283/15585924/09431156-2382-11e6-864f-36aea0ea3147.png)

1. GitTools
2. asbjornu
3. Pomona

I am a member of a lot more organizations than this, and HtmlUnit is one of them. Here's my GitHub profile, showing which organizations I'm a member of:

![screen shot 2016-05-26 at 20 43 47](https://cloud.githubusercontent.com/assets/12283/15586054/ac505c78-2382-11e6-828e-884f3d940300.png)

The weird red T-like icon there, is HtmlUnit. And as you can see, HtmlUnit is not listed in the dropdown on AppVeyor's login screen, so I can't log in to that account/organization on AppVeyor. If I log in to the "asbjornu" account, my personal GitHub projects are listed, which makes sense. I would also like to keep it that way. Renaming my personal "asbjornu" account to "HtmlUnit" to make it appear like the HtmlUnit organization sounds like a terrible hack and will take away my ability to run personal builds on AppVeyor.

Something is very weird here and it feels like #802 is something that really should be considered fixed so the confusion and mismatch between GitHub users & organizations and AppVeyor accounts & roles are remedied.

**Comment by asbjornu** [comment]
@FeodorFitsner This is still an issue for me. I am unable to log in under the HtmlUnit organization. Can you please assist?

**Comment by FeodorFitsner** [comment]
Nothing has been changed in AppVeyor's security model since that and this comment is still valid: https://github.com/appveyor/ci/issues/801#issuecomment-221948168

As I noted before AppVeyor's security model does not replicate GitHub's accounts/orgs structure. AppVeyor has its own accounts/users that can be associated with certain GH orgs/users.

**Comment by FeodorFitsner** [comment]
Sure, and sorry for the confusion it caused. We are definitely willing to improve that stuff over the time!

**Comment by asbjornu** [comment]
@FeodorFitsner Just to clear one thing up, now that I've created a build for [Schism Tracker](https://ci.appveyor.com/project/asbjornu/schismtracker); will a @schismtracker owner be able to create a new `schismtracker` project underneath a `schismtracker` user on AppVeyor? Or will that project forever be linked to my username?

**Comment by FeodorFitsner** [comment]
We've added a new section about creating an AppVeyor account for GitHub organization: https://www.appveyor.com/docs/team-setup/#setting-up-appveyor-account-for-github-organization - hope that helps.

**Comment by sschuberth** [comment]
While I'm happy to have these instructions, there's a big pitfall still as they say "On the same page authorize AppVeyor to access GitHub". If you do that while still being logged into GitHub with your usual account that is part of the GitHub org you want AppVeyor to build for, the AppVeyor org account will see *all* repos you have access to, not only the ones part of the specific GitHub org.

It seems the only way to work around this is to create a new GitHub user, give that user access to only the GitHub org you want to build for, and use that user's account to authorize AppVeyor.

Quite a pain if all you wanted in the beginning it to have builds for GitHub orgs available under and AppVeyor URL that includes the org's name instead of the user's name...

**Comment by asbjornu** [comment]
@sschuberth:

> It seems the only way to work around this is to create a new GitHub user, give that user access to only the GitHub org you want to build for, and use that user's account to authorize AppVeyor.

Yes, that's how I've "solved" this as well. It's unfortunate that this is necessary, but that's the only solution I've found for the problem.
